lib/Signaling/Listener.php

```diff
@@ -450,4 +452,27 @@ public static function notifyUsersViaExternalSignalingToRefreshTheChat(ChatEvent
 		];
 		$notifier->sendRoomMessage($room, $message);
 	}
+
+	public static function sendSignalingMessageWhenToggleRecording(ModifyRoomEvent $event): void {
+		$config = Server::get(Config::class);
+		$isSignalingDev = $config->isSignalingDev();
+
+		if (self::isUsingInternalSignaling() && $isSignalingDev) {
```

Comment:
Should this be `!$isSignalingDev`?

Reply:
I do not know :shrug: In fact I think that it should be `self::isUsingInternalSignaling() || $isSignalingDev` (note the `||`) so the message is not tried to be sent either when `isSignalingDev`. @vitormattos Could you clarify this? :-)

Related to that, I think that the conditions for [`isRecordingEnabled()`](https://github.com/nextcloud/spreed/blob/9cb74afbc48d587d406e316680c155686fca7ebd/lib/Config.php#L144-L153) should be changed. Although the official recording server requires the HPB I would not check the signaling server, as maybe other third-party recording servers could work without requiring the HPB (if I recall correctly _talked_ does not need an HPB). And independently of that, it should be checked that a recording server is configured.

Reply:
Hm, isn't the whole point of `$isSignalingDev` to circumvent the HPB requirement for development? So with your suggestion that would mean if I enabled signaling dev mode, no message is sent ever. Judging from this 
https://github.com/nextcloud/spreed/blob/9cb74afbc48d587d406e316680c155686fca7ebd/lib/Config.php#L147
and reversing the condition, I still think it should be `&&` and `!$isSignalingDev`. (Leaving the question aside you raised about limiting it to HPB at all).

Reply:
> So with your suggestion that would mean if I enabled signaling dev mode, no message is sent ever.

But the message can not be sent anyway with `$isSignalingDev`, no? If there is no real signaling server, [where is the request sent](https://github.com/nextcloud/spreed/blob/62fd48d1a764c24d731e24b9289917c6e2c9bdc8/lib/Signaling/BackendNotifier.php#L113)?

Reply:
Touché. I was under the impression, that it would then be sent through internal signaling, but that will fail at latest at https://github.com/nextcloud/spreed/blob/67088437d2de72417c37de36c6b359ac2986d029/lib/Signaling/BackendNotifier.php#L107-L110

Reply:
> Hm, isn't the whole point of $isSignalingDev to circumvent the HPB requirement for development?

It is.

Reply:
@nickvergessen Then shouldn't that either be 
```
self::isUsingInternalSignaling() || $isSignalingDev
```
(as Daniel suggested)
or
```
self::isUsingInternalSignaling() && !$isSignalingDev
```
Depending on what should be achieved here.

Reply:
The recording option should be visible/interactable when the `$config->isSignalingDev()` is set to true.
For actually sending I have to admit I don't care, because as outlined it will not make a functional difference?

Reply:
Right, but then I think we should remove the `isSignalingDev`-Check in `Signaling/Listener.php` to avoid any further confusion. This check is not added to any of the other methods